Experience Brand Kit 1.0

aaronmgdr opened the pull request:
### Description

A place to get assets / logos/ icons  for celo brand. A styleguide for Celo projects. 

This is the first part of the Experience tool kit. To start there are 4 pages. Logo, Typogrpahy, color, icons. 

##### Some Highlights. 

1. A Sidebar that responds to hash changes and scrolling 
2. Icons loading from airtable 
3. click/press to copy hexcode for colors
4. NO MORE TYPEKIT. now using google font and another third party for jost. 
5. futura is out jost is in. can you tell?

##### packages added 

- `yiq` 916b, tiny lib for  determining of color should have dark or light text over it. 
- ` colortranslator` 14k, for getting CMYK. not as small as id but didn't have time for smaller one.
-  "scroll-into-view":  previous scrolling intro view was not great. this one works much better than what i wrote

##### 
note a few links are still in TODO

### Tested

<img width="1695" alt="Screen Shot 2019-11-27 at 2 44 05 PM" src="https://user-images.githubusercontent.com/3814795/69758417-1874ee80-1125-11ea-89ee-faa43e637464.png">

#### Logo
<img width="1652" alt="Screen Shot 2019-11-27 at 2 44 33 PM" src="https://user-images.githubusercontent.com/3814795/69758418-1874ee80-1125-11ea-826f-0d48fa4ee770.png">
<img width="1397" alt="Screen Shot 2019-11-27 at 2 44 43 PM" src="https://user-images.githubusercontent.com/3814795/69758419-190d8500-1125-11ea-84f5-af137838271f.png">
<img width="1257" alt="Screen Shot 2019-11-27 at 2 44 51 PM" src="https://user-images.githubusercontent.com/3814795/69758420-190d8500-1125-11ea-8f2e-241947a23f99.png">
<img width="1238" alt="Screen Shot 2019-11-27 at 2 45 02 PM" src="https://user-images.githubusercontent.com/3814795/69758421-190d8500-1125-11ea-807c-727a7ab99d62.png">
<img width="1272" alt="Screen Shot 2019-11-27 at 2 45 10 PM" src="https://user-images.githubusercontent.com/3814795/69758422-190d8500-1125-11ea-966e-196979b9b461.png">
<img width="1271" alt="Screen Shot 2019-11-27 at 2 45 22 PM" src="https://user-images.githubusercontent.com/3814795/69758423-190d8500-1125-11ea-823d-a669cdc45d5f.png">
<img width="1259" alt="Screen Shot 2019-11-27 at 2 45 44 PM" src="https://user-images.githubusercontent.com/3814795/69758424-190d8500-1125-11ea-942e-835b3f16443f.png">
<img width="1189" alt="Screen Shot 2019-11-27 at 2 45 50 PM" src="https://user-images.githubusercontent.com/3814795/69758425-19a61b80-1125-11ea-88bc-4e310c5a20a5.png">

#### Color 
<img width="1298" alt="Screen Shot 2019-11-27 at 3 18 27 PM" src="https://user-images.githubusercontent.com/3814795/69760111-44926e80-1129-11ea-9945-f6e9d59ed61a.png">
<img width="1255" alt="Screen Shot 2019-11-27 at 3 18 32 PM" src="https://user-images.githubusercontent.com/3814795/69760112-452b0500-1129-11ea-9403-09f7553a87c6.png">
<img width="1276" alt="Screen Shot 2019-11-27 at 3 18 43 PM" src="https://user-images.githubusercontent.com/3814795/69760113-452b0500-1129-11ea-802a-04ac1b5a1119.png">
<img width="990" alt="Screen Shot 2019-11-27 at 3 18 51 PM" src="https://user-images.githubusercontent.com/3814795/69760114-452b0500-1129-11ea-8ce0-ff47d258c37c.png">

#### Typography
<img width="1083" alt="Screen Shot 2019-11-27 at 3 19 49 PM" src="https://user-images.githubusercontent.com/3814795/69760201-7c011b00-1129-11ea-98a0-503c38e21df6.png">
<img width="1062" alt="Screen Shot 2019-11-27 at 3 20 01 PM" src="https://user-images.githubusercontent.com/3814795/69760203-7c011b00-1129-11ea-8673-a23c7204fce6.png">
<img width="1055" alt="Screen Shot 2019-11-27 at 3 20 09 PM" src="https://user-images.githubusercontent.com/3814795/69760204-7c011b00-1129-11ea-960c-2866092d3308.png">
<img width="1076" alt="Screen Shot 2019-11-27 at 3 20 16 PM" src="https://user-images.githubusercontent.com/3814795/69760205-7c99b180-1129-11ea-9da5-5480ea3cb230.png">


#### Icons
<img width="1656" alt="Screen Shot 2019-11-27 at 3 13 07 PM" src="https://user-images.githubusercontent.com/3814795/69759775-87a01200-1128-11ea-9fd2-09581e946b83.png">
<img width="1651" alt="Screen Shot 2019-11-27 at 3 13 17 PM" src="https://user-images.githubusercontent.com/3814795/69759776-8838a880-1128-11ea-9151-8515d102019d.png">


### Other changes

finally replace all the adhoc Links...

```
function Link({ children, href }: Props) {
  return <Button text={children} href={href} kind={BTN.INLINE} />
}
```

..which I had been creating randomly whenever A link was needed to be interpolated into a translated text. 

Now there is one `InlineAnchor` component 

### Related issues

- see https://www.figma.com/file/HAe97Ol6f1O0suiF1oE83Z/Brand-Kit-1.0?node-id=0%3A1

### Backwards compatibility

brand new!

aaronmgdr committed:
Steal this from other branch

aaronmgdr committed:
Static version of Menu us looking good

aaronmgdr committed:
Brand kit general layout

aaronmgdr committed:
Add base pages and other fun stuff

aaronmgdr committed:
Wip

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/brandkit

* master:
  Adjust e2e transfer and governance tests to match new fee distribution and eliminate ProposerFraction (#1585)
  [Wallet] Add more local currencies (#1698)
  Switch to correct cluster when fauceting (#1687)
  [Wallet] Use the country of the phone number for determining the default local currency (#1684)
  [Wallet] Limit QR code scanner to 1 code per second (#1676)
  Update Dark backgrounds text color (#1677)
  Remove integration sync test
  Minor attestation service fixes (#1680)
  Add golang to setup docs

aaronmgdr committed:
Section nav

aaronmgdr committed:
Redo pages

aaronmgdr committed:
Sane scroll behavior

aaronmgdr committed:
Use constants for all paths

aaronmgdr committed:
Mobile

aaronmgdr committed:
WIP fetching from air table

aaronmgdr committed:
Logo page basic content first pass

aaronmgdr committed:
Add in words (WIP)

aaronmgdr committed:
Better white spacing

aaronmgdr committed:
Move some files around and other things

aaronmgdr committed:
Ensure long menu would be scrollable

aaronmgdr committed:
Colors true

aaronmgdr committed:
no message

aaronmgdr committed:
Typeography Page A

aaronmgdr committed:
Fetching icons from airtbale and and displaying on page nice

aaronmgdr committed:
Usage Example

aaronmgdr committed:
Fix ilocal keys

aaronmgdr committed:
Right on

aaronmgdr committed:
Type overhaul w han

aaronmgdr committed:
Judement

aaronmgdr committed:
Improve Logo page on mobile

aaronmgdr committed:
LOGOGOSOGOOGOS

aaronmgdr committed:
Change URI to be /experience/brand/*

aaronmgdr committed:
LINT this and add brand redirect

aaronmgdr committed:
Icons view change and fix

aaronmgdr committed:
Create download button, fix up nav and make font page great

aaronmgdr committed:
Remove board and left alignment on colors

aaronmgdr committed:
Better gap

aaronmgdr committed:
Alignment of top

aaronmgdr committed:
Color Page Content

aaronmgdr committed:
Create Inline Anchor for using inline button in the trans
Remove hash hoc

Update general cleanup

aaronmgdr committed:
Brand home page is a ok

aaronmgdr committed:
Logo page content and better fit on tablet

aaronmgdr committed:
Logo page content
Some sizing stuff

Ensure there is a minimum size for sidebar

cmcewen committed:
Add web deps

aaronmgdr committed:
Tyopgraphy page

aaronmgdr committed:
ICONS text

aaronmgdr committed:
Beautiful icons

aaronmgdr committed:
Sweet Jesus its downloading season

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/brandkit

* master: (97 commits)
  Fix celotool expected cluster check (#1932)
  Build Page for Baklava Launch Part 1  (#1866)
  cli: Fix for #1875 (#1920)
  cli: Fixes for #1880 and #1874 (#1918)
  Minor edits to full-node and other docs (#1917)
  Check that we have provisioned phone numbers (#1927)
  Sort invite contact list (#1936)
  Generate the pop for the attestation service with geth (#1931)
  Have validators use ephemeral node keys by default (#1922)
  Enhanced Validator Setup Docs (#1926)
  Indicate account registration (#1928)
  Bugfix: actually whitelist oracle account (#1865)
  Display current account locked gold requirement in lockedgold:show (#1923)
  Dedicated signing of attestation messages (#1913)
  Allow BLS info to be passed to validator:register without 0x prefix (#1919)
  Use cert manager instead of lego for k8s SSL certificates (#1924)
  Fix protocol tests (#1835)
  Fix bug in LockedGold.relock wrapper (#1915)
  Add account:show command (#1916)
  Add warning about misusing pumba (#1857)
  ...

# Conflicts:
#	packages/web/pages/_app.tsx
#	yarn.lock

aaronmgdr commented on packages/web/src/brandkit/common/Page.tsx:
```diff
@@ -0,0 +1,240 @@
+
+const TYPE_PATH = `${ROOT}/typography`
+
+// const IMAGERY_PATH = `${ROOT}/key-imagery`
```
coming in v1.1

aaronmgdr commented on packages/web/src/brandkit/common/Page.tsx:
```diff
@@ -0,0 +1,240 @@
+    href: ICONS_PATH,
+    sections: [],
+  },
+  // {
```
imagery will return in v1.1

cmcewen replied:
to reduce comments, maybe just define this object outside the array and then don't use it? (might have to ts-ignore)

aaronmgdr replied:
if i was gonna do that id just delete it, plan is to have version with imagery page in up by end of the week.

aaronmgdr commented on packages/web/src/shared/Hoverable.tsx:
```diff
@@ -95,12 +97,13 @@ export default class Hoverable extends React.Component<Props, State> {
 
+    const child = getChild(children, isHovering)
+
+    // @ts-ignore
```
even with my fancy getChild function which does type checking it still wasnt sure about the type

aaronmgdr commented on packages/web/src/styles.ts:
```diff
@@ -94,6 +106,15 @@ export const fonts = StyleSheet.create({
+    textRendering: 'geometricPrecision',
+  },
+  // deprecated replaced with h6
   h5: {
```
I didnt want to cause the diff on this PR to be even larger by changing all font.h5 refs to fonts.h6. i think this can be done separately

aaronmgdr commented on packages/web/src/styles.ts:
```diff
@@ -24,25 +23,37 @@ export enum colors {
+  garamond = 'EB Garamond, eb-garamond, Garamond, serif',
 }
 
-export const fonts = StyleSheet.create({
```
change this so that on typography page i could print the styles out and know for sure that they matched. DRY

aaronmgdr committed:
Self review

cmcewen committed:
Merge branch 'cmcewen/web-fix' into aaronmgdr/brandkit

cmcewen committed:
Fix import

cmcewen committed:
Merge branch 'master' into aaronmgdr/brandkit

cmcewen commented on packages/web/server/AssetBase.ts:
```diff
@@ -0,0 +1,67 @@
+const TERMS_SIGNED = 'Terms=1'
+
+function normalize(asset: Fields) {
+  console.warn(asset)
```
did you mean to leave this?

cmcewen commented on packages/web/src/brandkit/Color.tsx:
```diff
@@ -0,0 +1,150 @@
+  const Component = image ? ImageBackground : View
+
+  return (
+    // @ts-ignore
```
What's the error here?

aaronmgdr replied:
Component does not have call signature ?

cmcewen commented on packages/web/src/brandkit/IconsPage.tsx:
```diff
@@ -0,0 +1,77 @@
+    <Page
+      sections={[
+        {
+          id: 'overview',
```
maybe use that hashNav?

cmcewen commented on packages/web/src/brandkit/IconsPage.tsx:
```diff
@@ -0,0 +1,77 @@
+  uri: string
+}
+
+const LOADING = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
```
if you want to be fancy: https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

cmcewen commented on packages/web/src/brandkit/KeyImagery.tsx:
```diff
@@ -0,0 +1,38 @@
+import * as React from 'react'
+import { withNamespaces } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+// import Fetch from 'src/brandkit/common/Fetch'
```
comment

cmcewen commented on packages/web/src/brandkit/KeyImagery.tsx:
```diff
@@ -0,0 +1,38 @@
+
+const { brandImagery } = hashNav
+
+// TODO in v 1.1
```
does this mean this file isn't used? or what is TODO?

cmcewen commented on packages/web/src/brandkit/Typography.tsx:
```diff
@@ -0,0 +1,162 @@
+import { fontInfo, fonts, standardStyles } from 'src/styles'
+const { brandTypography } = hashNav
+
+function getWeight(number) {
```
this is one where types would be a little helpful

cmcewen commented on packages/web/src/brandkit/common/Fetch.tsx:
```diff
@@ -0,0 +1,36 @@
+  }
+
+  componentDidMount = async () => {
+    const response = await fetch(this.props.query)
```
do you need to try catch this?

cmcewen commented on packages/web/src/brandkit/common/Page.tsx:
```diff
@@ -0,0 +1,242 @@
+    }
+  }
+
+  observation = () => {
```
same here - observeHeaderScroll, or something like that so that i know the purpose of this is to update the hash

cmcewen commented on packages/web/src/brandkit/common/Page.tsx:
```diff
@@ -0,0 +1,242 @@
+    this.setState({ routeHash: window.location.hash })
+  }
+
+  onIntersection = (entries: IntersectionObserverEntry[]) => {
```
It took me a while to understand what was going on here because the functions are more named on "how" something is happening than "what". the purpose of this is to update the window hash as well as the sidebar - so maybe something like updateCurrentSection or something like that rather than me having to figure out what the intersection is about

aaronmgdr replied:
for sure, there are a lot of functions needed for observation intersection to do what it needs to do. 
took me a while to get it figure out too
weill name more semantically

aaronmgdr committed:
Dont show border on last type example
Add margin to subtitles
Make name Celo White
Fix monochrome logo

aaronmgdr committed:
Glphs on backgrounds should be square on mobile

Icons make square small enough for 2x2 on mobile

Keep squares square and variable size

Add logo downloads

Thicken stroke on download icon

Remove buttons example

Converte minimums to be sag not png for better crispy scale

Better mobile spacing in general

Add new uiSmall first class type style

aaronmgdr committed:
Lint

aaronmgdr committed:
Lint

aaronmgdr committed:
Cleanup from PR

aaronmgdr committed:
Ensure the sidebar is not ever covered by the footer

aaronmgdr committed:
Add License

aaronmgdr committed:
Super duper with Laurance

aaronmgdr committed:
Lint

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/brandkit

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/brandkit

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/brandkit
